File: leetcode/206.c
#include <stdlib.h>

struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode *reverseList(struct ListNode *head) {
  struct ListNode *result = NULL;
  struct ListNode *tmp = NULL;

  while (head != NULL) {
    tmp = head->next;
    head->next = result;
    result = head;
    head = tmp;
  }

  return result;
}

File: leetcode/206.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        result = None
        tmp = None

        while head is not None:
            tmp = head.next
            head.next = result
            result = head
            head = tmp
        
        return result